src/core/option.rs

```diff
@@ -162,7 +162,7 @@ impl<T> Core<T> for OptionCore<T> {
     unsafe fn has_element_at(&self, idx: usize) -> bool {
         debug_assert!(idx < self.cap());
 
-        self.data.get_unchecked(idx).is_some()
+        idx < self.len() && self.data.get_unchecked(idx).is_some()
```

Comment:
I think perhaps this should instead be checked in the safe function `StableVecFacade::has_element_at` by changing the following check to `self.len()`, however running into some miri failures under tree-borrows with that change in place.
https://github.com/LukasKalbertodt/stable-vec/blob/7e6d29cea532c76a4a5d196c71e9f4edc07e5596/src/lib.rs#L550

Reply:
I think that comment is wrong and the patch as is probably right as-is in that it seems to make it match the behavior of the other `*Core` implementations, but those can implement `has_element_at` in a way that don't cause an uninitialized read.

Reply:
Yeah, I made a long attempt at that, between those two comments, the main thing is there are a lot of other cases mostly the `find_empty_slot*` ones that also call `has_element_at`, and so all those need to be updated too.
It much more involved at least than I had initially imagined in the first comment.